### ISSUE TITLE
NPE after deleting a MatParam 2x

### ISSUE THREAD

**Issue by stephengold** [stephengold/Maud#6]
At commit: 4cd137d5
To reproduce: 6, 1, Select geometry, 1, Select parameter, 1, Delete, Z, Delete
Expected: darker model with fewer material parameters
Got:
```text

Sep 07, 2021 9:22:55 AM jme3utilities.nifty.BasicScreenController perform
SEVERE: Caught unexpected throwable:
java.lang.NullPointerException
	at maud.model.cgm.MatParamRef.<init>(MatParamRef.java:90)
	at maud.model.cgm.SelectedMatParam.makeRef(SelectedMatParam.java:230)
	at maud.model.cgm.SelectedMatParam.delete(SelectedMatParam.java:78)
	at maud.action.DeleteAction.process(DeleteAction.java:92)
```

Hypothesis: cloning bug in `Material` or `MatParam`

Note that after the undo, the parameter type and value are both displayed as "null"! They should be "Vector4" and "Color[0.7058824, 0.6509804, 0.6, 1.0]" respectively.

**Comment by stephengold** [comment]
After the undo, `getMatParam().findNameIndex()` [which uses `material.getParams()`] returns a negative value, yet `getMatParam().isSelected()` [which looks only at `selectedName`] returns true.

**Comment by stephengold** [comment]
The `paramValues` map in `Material` has only 8 entries and does not contain one for "Ambient". `Material` is `Cloneable` but not `JmeCloneable`. `Material.clone()` is invoked during the MP delete (both for Jaime and SkyControl) and appears to clone `paramValues` correctly.

**Comment by stephengold** [comment]
Note that `History.undo()` invokes `Cgm.clone()` 4x: cloning source and target for both `new Checkpoint` and `previous.restore()`. The issue is presumably during `restore()` of the target.

**Comment by stephengold** [comment]
When the first checkpoint is created, it reports the `VarType` as `Vector4`. The same checkpoint is later restored, but at that time it reports the `VarType` as `null`. In fact, it's the same `SelectedMatParam` instance and the same `Material` instance from when the checkpoint was created. Looks like the `Material` got aliased somewhere.

**Comment by stephengold** [comment]
`Geometry.cloneFields()` sure looks like it clones the `material` field...

**Comment by stephengold** [comment]
Looks like `Cgm.clone()` fails to update a few backpointers. This might be an easy fix!

**Comment by stephengold** [comment]
Fixed in "master" at 5aff82ea

Well, that kept me entertained for a few hours...
